src/vehicle.cpp

```diff
@@ -3381,13 +3387,23 @@ void vehicle::consume_fuel( double load = 1.0 )
 
         //get exact amount of fuel needed
         double amnt_precise = double(amnt_fuel_use) / ft.coeff;
-
         amnt_precise *= load * (1.0 + st * st * 100);
+        int remainder = int( amnt_precise * 1000 ) % 1000;
         int amnt = int(amnt_precise);
-        // consumption remainder results in chance at additional fuel consumption
-        if( x_in_y(int(amnt_precise*1000) % 1000, 1000) ) {
-            amnt += 1;
+
+        if( remainder != 0 ) {
+            auto previous_remainder = fuel_remainder.find( ft.id );
+            if( previous_remainder != fuel_remainder.end() ) {
+                previous_remainder->second += remainder;
+		if( previous_remainder->second >= 1000 ) {
+                    previous_remainder->second -= 1000;
+                    amnt++;
```

Comment:
That's exactly what happens.  Previously the code would calculate an amount and a remainder, and "randomly" increment the amount based on how big the remainder was.  Concretely, if amnt_precise was 2.5, it had a 50% chance of setting amnt to 3.
Now it will instead stash the 500, and add it to the next iteration, if it's 2.5 again, it will sum to 1000 and amnt is incremented to 3 and remainder is reset.
Now that I think about it, remainder should be stored as a float.